src/linker/Linker/MarkingReason.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Mono.Linker
+{
+	//
+	// Try to keep the name short and self explamantory at the same time. The name
+	// is stored in the output format as text and be better easily understood.
+	//
+	public enum MarkingReason
+	{
+		//
+		// Entry points to the analysis
+		//
+		AssemblyAction = 1,
```

Comment:
Can we keep the comments from the original enum? I like that those include what is the edge the reason is used on - it makes it easier to understand the reason.